### PR TITLE
Compliance tests

### PR DESCRIPTION
**Pull request by TomasArrachea**
### What Changed?
This PR works on compliance tests to compare sdk status agains other implementations (like [rust-sdk](https://github.com/Layr-Labs/eigensdk-rs))

Modified 2 tests (`TestAvsRegistryServiceChainCaller_GetOperatorsAvsState` and `TestIntegrationBlsAgg`) to read the input from a JSON file in the case that the `TEST_DATA_PATH` env variable is set.

### Reviewer Checklist
- [ ] Code is well-documented
- [ ] Code adheres to Go [naming conventions](https://go.dev/doc/effective_go#names)
- [ ] Code deprecates any old functionality before removing it

### CODE REVIEW

File: testutils/test_data.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"os"
+)
+
+// Test data for generic loading of JSON data files.
+type TestData[T any] struct {
+	Input T `json:"input"`
+}
+
+// Create a new instance of `TestData` with the given input data.
+func NewTestData[T any](defaultInput T) TestData[T] {
```
Comment (shrimalmadhur) [inline testutils/test_data.go:15]: is there any test file we should add to test this?

Comment (TomasArrachea) [reply testutils/test_data.go:15]: The idea is to run these compliance tests from a separate framework, the test files will be set in [that repo](https://github.com/Layr-Labs/eigensdk-compliance).

Comment (shrimalmadhur) [reply testutils/test_data.go:15]: oh interesting - make ssense

**Review by shrimalmadhur — APPROVED** [review]
LGTM. Is it ready to be merged?

**Comment by TomasArrachea** [comment]
> LGTM. Is it ready to be merged?

@shrimalmadhur Yes!

**Pull request merged**